sql/log.cc

```diff
@@ -7992,23 +8008,25 @@ MYSQL_BIN_LOG::write_transaction_to_binlog_events(group_commit_entry *entry)
 {
   int is_leader= queue_for_group_commit(entry);
 #ifdef WITH_WSREP
-  if (wsrep_is_active(entry->thd) &&
-      wsrep_run_commit_hook(entry->thd, entry->all))
+  /* commit order was released in queue_for_group_commit() call,
+     here we check if wsrep_commit_ordered() failed or if we are leader */
+  switch (is_leader)
   {
-    /*
-      Release commit order and if leader, wait for prior commit to
-      complete. This establishes total order for group leaders.
-    */
-    if (wsrep_ordered_commit(entry->thd, entry->all))
-    {
-      entry->thd->wakeup_subsequent_commits(1);
-      return 1;
-    }
-    if (is_leader)
-    {
-      if (entry->thd->wait_for_prior_commit())
-        return 1;
-    }
+  case -2: /* wsrep_ordered_commit() has failed */
+    entry->thd->wakeup_subsequent_commits(1);
+    return true;
+  case -3: /* this is leader, wait for prior commit to
+              complete. This establishes total order for group leaders
+           */
+    if (entry->thd->wait_for_prior_commit())
+      return true;
```

Comment:
Both these cases are missing
```c++
DBUG_ASSERT(wsrep_is_active(entry->thd));
```
If `wsrep_run_commit_hook(entry->thd, entry->all)` does not have side effects, then there should be another assertion for that one.